api/v1beta1/common_types.go

```diff
@@ -112,120 +112,20 @@ type BarbicanComponentTemplate struct {
-	// HMAC Keygen Mechanism
-	HMACKeygenMechanism string `json:"HMACKeygenMechanism"`
-
-        // +kubebuilder:validation:Optional
-	// +kubebuilder:default=CKM_SHA256_HMAC
-	// HMAC Mechanism. This replaces hsm_keywrap_mechanism
-	HMACMechanism string `json:"HMACMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=4
-        // +kubebuilder:validation:Maximum=7
-        // +kubebuilder:validation:Minimum=0
-        // Level of logging, where 0 means "no logging" and 7 means "debug".
-        LoggingLevel int `json:"loggingLevel"`
-
-	// +kubebuilder:validation:Required
-	// The HSM's IPv4 address (X.Y.Z.K)
-	ServerAddress string `json:"serverAddress"`
-
-	// +kubebuilder:validation:Optional
-	// The IP address of the client connecting to the HSM (X.Y.Z.K)
-	ClientAddress string `json:"clientAddress,omitempty"`
-
         // +kubebuilder:validation:Required
         // OpenShift secret that stores the password to login to the PKCS11 session
         LoginSecret string `json:"loginSecret"`
 
-        // +kubebuilder:validation:Optional
-        // The OpenShift secret that stores the HSM certificates.
-        CertificatesSecret string `json:"certificatesSecret,omitempty"`
-
-        // +kubebuilder:validation:Optional
-        // The mounting point where the certificates will be copied to (e.g., /usr/local/luna/config/certs).
-	CertificatesMountPoint string `json:"certificatesMountPoint,omitempty"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=CKM_AES_GCM
-        // Secret encryption mechanism
-	EncryptionMechanism string `json:"encryptionMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=CKM_AES_KEY_WRAP_KWP
-        // Key wrap mechanism
-	KeyWrapMechanism string `json:"keyWrapMechanism"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=true
-        // Generate IVs for the key wrap mechanism
-	KeyWrapGenerateIV bool `json:"keyWrapGenerateIV"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=true
-        // Generate IVs for CKM_AES_GCM mechanism
-	AESGCMGenerateIV bool `json:"AESGCMGenerateIV"`
-
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=true
-        // Always set cka_sensitive
-	AlwaysSetCKASensitive bool `json:"alwaysSetCKASensitive"`
+        // +kubebuilder:validation:Required
+        // The OpenShift secret that stores the HSM client data.
+        // These will be mounted to /etc/hsm-client
+        ClientDataSecret string `json:"clientDataSecret"`
 
-        // +kubebuilder:validation:Optional
-        // +kubebuilder:default=false
-        // Set os_locking_ok
-	OSLockingOK  bool `json:"OSLockingOK"`
+        // +kubebuilder:validation:Required
+        // Location to which kolla will copy the data in ClientDataSecret.
+        ClientDataPath string `json:"clientDataPath"`
```

Comment:
wondering if we could make it optional and default it to e.g. `/etc/hsm-client`? see my comment on the barbican kolla file. would remove the req to have the condition in it.

Reply:
Done